src/sources/prometheus/parser.rs

```diff
@@ -1167,4 +1181,129 @@ mod test {
             ]),
         );
     }
+
+    macro_rules! map {
+        ($($key:expr => $value:expr),+) => {
+            {
+                let mut m = ::std::collections::BTreeMap::new();
+                $(
+                    m.insert($key.into(), $value.into());
+                )+
+                m
+            }
+        };
+    }
+
+    // https://github.com/timberio/vector/issues/3276
+    #[test]
+    fn test_nginx() {
+        let exp = r##"
+            # HELP nginx_server_bytes request/response bytes
+            # TYPE nginx_server_bytes counter
+            nginx_server_bytes{direction="in",host="*"} 263719
+            nginx_server_bytes{direction="in",host="_"} 255061
+            nginx_server_bytes{direction="in",host="nginx-vts-status"} 8658
+            nginx_server_bytes{direction="out",host="*"} 944199
+            nginx_server_bytes{direction="out",host="_"} 360775
+            nginx_server_bytes{direction="out",host="nginx-vts-status"} 583424
+            # HELP nginx_server_cache cache counter
+            # TYPE nginx_server_cache counter
+            nginx_server_cache{host="*",status="bypass"} 0
+            nginx_server_cache{host="*",status="expired"} 0
+            nginx_server_cache{host="*",status="hit"} 0
+            nginx_server_cache{host="*",status="miss"} 0
+            nginx_server_cache{host="*",status="revalidated"} 0
+            nginx_server_cache{host="*",status="scarce"} 0
+            "##;
```

Comment:
@binarylogic The test case you mentioned, it doesn't fail though.

Reply:
Hm, that's strange. I'd be curious to see if it fails during an actual integration test with the `prometheus` source. Seeing that we don't have an integraiton test for this source I've opened #3289. What do you think?

Reply:
It would be great if we can get more details about the error (via #3173).